atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceTestUtils.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    public static ImmutableMap<TableReference, byte[]> createByteArrayByTableReferenceMapWithSize(int size) {
+        return ImmutableMap.of(
+                createTableReferenceWithSize(size / 6, 'A'), createBytesWithSize(size / 6, (byte) 0),
+                createTableReferenceWithSize(size / 6, 'B'), createBytesWithSize(size / 6, (byte) 1),
+                createTableReferenceWithSize(size / 6, 'C'), createBytesWithSize(size - 5 * (size / 6), (byte) 2));
+    }
+
+    public static ImmutableSet<TableReference> createTableReferenceSetWithSize(int size) {
+        return ImmutableSet.of(
+                createTableReferenceWithSize(size / 3, 'A'),
+                createTableReferenceWithSize(size / 3, 'B'),
+                createTableReferenceWithSize(size - 2 * (size / 3), 'C'));
+    }
+
+    public static ImmutableList<RowResult<Value>> createValueRowResultListWithSize(int size) {
+        return ImmutableList.of(
+                createValueRowResultWithSize(size / 3, (byte) 0),
+                createValueRowResultWithSize(size / 3, (byte) 1),
+                createValueRowResultWithSize(size - 2 * (size / 3), (byte) 2));
+    }
+
+    public static ImmutableList<RowResult<Set<Long>>> createLongSetRowResultListWithSize(int size) {
+        return ImmutableList.of(
+                createLongSetRowResultWithSize(size / 3, (byte) 0),
+                createLongSetRowResultWithSize(size / 3, (byte) 1),
+                createLongSetRowResultWithSize(size - 2 * (size / 3), (byte) 2));
+    }
+
+    public static ImmutableList<ImmutableList<CandidateCellForSweeping>> createCandidateCellForSweepingTableWithSize(
+            int size) {
+        return ImmutableList.of(
+                createCandidateCellForSweepingList(size / 3, (byte) 0),
+                createCandidateCellForSweepingList(size / 3, (byte) 1),
+                createCandidateCellForSweepingList(size - 2 * (size / 3), (byte) 2));
+    }
+
+    public static ImmutableMap<Cell, Value> createValueByCellMapWithSize(int size) {
+        return ImmutableMap.of(
+                createCellWithSize(size / 6, (byte) 0), createValueWithSize(size / 6),
+                createCellWithSize(size / 6, (byte) 1), createValueWithSize(size / 6),
+                createCellWithSize(size / 6, (byte) 2), createValueWithSize(size - 5 * (size / 6)));
+    }
+
+    public static ImmutableMap<Cell, Long> createLongByCellMapWithSize(int size, long identifier) {
+        int keySetSize = size - 3 * Long.BYTES;
+        return ImmutableMap.of(
+                createCellWithSize(keySetSize / 3, (byte) 1), identifier,
+                createCellWithSize(keySetSize / 3, (byte) 2), identifier,
+                createCellWithSize(keySetSize - 2 * (keySetSize / 3), (byte) 3), identifier);
+    }
+
+    @SuppressWarnings("ArrayAsKeyOfSetOrMap")
+    public static Map<byte[], RowColumnRangeIterator> createRowColumnRangeIteratorByByteArrayMutableMapWithSize(
```

Comment:
should probably make this an identity hash map - suppressing this exception is not the right thing to do.